kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/oxford/WearableCameraConverterFactory.kt

```diff
@@ -8,46 +8,58 @@ import org.radarbase.connect.upload.converter.FileProcessorFactory
 import org.radarbase.connect.upload.converter.LogRepository
 import org.radarbase.connect.upload.converter.ZipFileProcessorFactory
 import org.radarbase.connect.upload.io.FileUploader
+import org.radarbase.connect.upload.io.LocalFileUploader
 import org.radarbase.connect.upload.io.SftpFileUploader
 import org.slf4j.LoggerFactory
 import java.net.URI
-import java.net.URL
+import java.nio.file.Path
 import java.nio.file.Paths
 
 class WearableCameraConverterFactory : ConverterFactory {
     override val sourceType: String = "oxford-wearable-camera"
 
-    private val sftpUploader = ThreadLocal<FileUploader>()
+    private val localUploader = ThreadLocal<FileUploader>()
 
     override fun fileProcessorFactories(settings: Map<String, String>, connectorConfig: SourceTypeDTO, logRepository: LogRepository): List<FileProcessorFactory> {
         val sourceConfig = checkNotNull(connectorConfig.configuration)
-        val credentials = SftpFileUploader.SftpCredentials(
-                host = sourceConfig.getValue("host"),
-                port = sourceConfig["port"]?.toInt() ?: 22,
-                username = sourceConfig.getValue("user"),
-                password = sourceConfig["password"],
-                privateKeyFile = sourceConfig["keyFile"],
-                privateKeyPassphrase = sourceConfig["keyPassphrase"])
 
-        val root = Paths.get(sourceConfig["root"] ?: ".")
+        val advertizedUrl: URI
+        val uploaderSupplier: () -> FileUploader
+        val root: Path
 
-        var urlString = sourceConfig["advertizedUrl"] ?: "sftp://${credentials.host}:${credentials.port}"
-        if (!urlString.endsWith("/")) urlString += "/"
-        val advertizedUrl = URI.create(urlString)
+        if ("host" in sourceConfig) {
+            val credentials = SftpFileUploader.SftpCredentials(
```

Comment:
Can we add two lines of INFO or debug line to log that the data is written to sftp or localfilesystem

Reply:
Done